src/IntegrationUtils.ts

```diff
@@ -181,9 +201,52 @@ function getSelectedResource() {
 		}
 	});
 }
+		
+function getSelectedProperties() {
+	return new Promise <any> ( async (resolve, reject) => {
+		let hasMoreProperties: boolean = true;
+		let returnedProperties : string[] = [];
+		while (hasMoreProperties) {
+			let newProperty = undefined;
+			await vscode.window.showInputBox({
+				placeHolder: 'Specify the property name',
+				ignoreFocusOut: true,
+				validateInput: validateName
+			}).then ( async (propName) => {
+				await vscode.window.showInputBox({
+					placeHolder: 'Specify the property value',
+					ignoreFocusOut: true
+				}).then ( (propValue) => {
+					if (propValue) {
+						propValue = propValue.replace(/"/g, '\\"');
+						newProperty = `${propName}="${propValue}"`;
+					}
+				});
+			});
+			if (!newProperty) {
+				reject(new Error(`No Property provided`));
+				return;
```

Comment:
this return looks obsolete

Reply:
Though I understand that it looks that way, I am finding that it doesn't always work that way. For some reason, rejecting doesn't kick out of the method. So I just ensure that it does by putting in the return

Reply:
but there is no more code after this? Just the else case.

Reply:
the if handles the case where the user cancels out. the else handles the case where we then check the answer, which they can also cancel out of before we get an answer

Reply:
Cleaned up